### PR TITLE
fix 'Client scripts: test's client scripts rewrite fixture's client scripts' (close #4122)

### LINKED ISSUE #4122: Client scripts: test's client scripts rewrite fixture's client scripts
import { ClientFunction } from 'testcafe';

fixture `Fixture`
    .clientScripts({ content: 'var i = 0;'});

test('test', async t => {
    await t.expect(ClientFunction(() => window['i'])()).eql(1);
}).clientScripts({ content: 'var j = 1;'});

### PR DESCRIPTION
**Pull request by miherlosev**
@helen-dikareva @AndreyBelym

### CODE REVIEW

File: test/functional/fixtures/api/es-next/custom-client-scripts/testcafe-fixtures/i4122.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ClientFunction } from 'testcafe';
+
+fixture `Fixture`
+    .clientScripts({ content: 'window["test"] = 0;' });
+
+const getTestValue = ClientFunction(() => window['test']);
+
+test('test', async t => {
+    await t.expect(getTestValue()).eql(1);
```
Comment (helen-dikareva) [inline test/functional/fixtures/api/es-next/custom-client-scripts/testcafe-fixtures/i4122.js]: This test doesn't check that both scripts are injected, does it?

Comment (miherlosev) [reply test/functional/fixtures/api/es-next/custom-client-scripts/testcafe-fixtures/i4122.js]: Yes, it is. I can rewrite this.

Comment (helen-dikareva) [reply test/functional/fixtures/api/es-next/custom-client-scripts/testcafe-fixtures/i4122.js]: let's rewrite it little bit

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/api/structure/test.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ export default class Test extends TestingUnit {
 
         assertRequestHookType(hooks);
 
-        this.requestHooks = hooks;
+        this.requestHooks = union(this.requestHooks, hooks);
```
Comment (AndreyBelym) [inline src/api/structure/test.js]: It changes the method's behavior. Before this change, each call to `test.requestHooks()` overrode the previous value. Now additional calls add additional hooks/scripts.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**